polyfill.js

```diff
@@ -59,15 +59,46 @@
     }
 
     /**
-     * @param {TypedArray} species
+     * @param {TypedArray} arr
+     * @returns {TypedArray[typeof Symbol.toStringTag]}
+     */
+    function typedArrayNameInternalSlot(arr) {
+        return Object.getOwnPropertyDescriptor(typedArrayPrototype, Symbol.toStringTag)
+            .get.call(arr);

```

Comment:
Cute trick! I didn't realize that this would work, but it does.

Reply:
yep! For a moment I was worried there wouldn't be a way for polyfills to get access to the slot.